### PR TITLE
Add entities (Atrem loh)

### PR DESCRIPTION
**Pull request by Mrjoulin**

### CODE REVIEW

File: stage-4/database-course/src/main/kotlin/su/arlet/entities/Batch.kt [hunk 1]
```diff
@@ -0,0 +1,1964 @@
...
+    val packsCount = integer("packs_count").nullable()
+        .check("positive_count") { it.isNull().or(it.greaterEq(0)) }
+    val massOfDefective = double("mass_of_defective").nullable()
+        .check("positive_defective_mass") { it.isNull().or(it.greaterEq(0.0)) }
+
+    override val primaryKey = PrimaryKey(id)
+}
+
+
+// .
```
Comment (Arlet2) [inline stage-4/database-course/src/main/kotlin/su/arlet/entities/Batch.kt:45]: Remove waves, please

Comment (Mrjoulin) [reply stage-4/database-course/src/main/kotlin/su/arlet/entities/Batch.kt:45]: No.

File: stage-4/database-course/src/main/kotlin/su/arlet/entities/ConveyorPelmeni.kt [hunk 2]
```diff
@@ -0,0 +1,1984 @@
...
+import org.jetbrains.exposed.sql.*
+
+
+data class ConveyorPelmeni (
+    val conveyorId: Int,
+    val typeId: Int,
+    val sizeName: String
+)
+
+object ConveyorPelmenies : Table("conveyor_pelmeni") {
```
Comment (Arlet2) [inline stage-4/database-course/src/main/kotlin/su/arlet/entities/ConveyorPelmeni.kt:12]: I think we don't need association tables in ORM. Need to discuss

File: stage-4/database-course/src/main/kotlin/su/arlet/entities/FreezingInfo.kt [hunk 3]
```diff
@@ -0,0 +1,1990 @@
...
+
+data class FreezingInfoObject (
+    val batchID: Int,
+    val sectionID: Int,
+    val temperature: Double,
+    val loadDate: LocalDate,
+    val uploadDate: LocalDate?
+)
+
+object FreezingInfo : Table("freezing_info") {
```
Comment (Arlet2) [inline stage-4/database-course/src/main/kotlin/su/arlet/entities/FreezingInfo.kt:16]: Also as conveyour pelmeni problem

File: stage-4/database-course/src/main/kotlin/su/arlet/entities/PelmeniType.kt [hunk 4]
```diff
@@ -0,0 +1,1964 @@
...
+
+
+data class PelmeniType(
+    val typeID: Int,
+    val name: String,
+    val recipe: String
+)
+
+object PelmeniTypes : Table("pelmeni_types") {
+    val typeID = integer("type_id").autoIncrement()
```
Comment (Arlet2) [inline stage-4/database-course/src/main/kotlin/su/arlet/entities/PelmeniType.kt:13]: typeID -> id

File: stage-4/database-course/src/main/kotlin/su/arlet/entities/WorkShift.kt [hunk 5]
```diff
@@ -0,0 +1,1974 @@
...
+
+import java.time.LocalDateTime
+
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.javatime.timestamp
+
+
+data class WorkShift(
+    val id: Int,
+    val batchID: Int?,
```
Comment (Arlet2) [inline stage-4/database-course/src/main/kotlin/su/arlet/entities/WorkShift.kt:11]: why nullable if we have one2one relation

**Comment by Arlet2** [comment]
Exposed in Kotlin is not providing any special syntax for m2m references, you did it in Exposed ideomatic way. Just ignore comments about assocations.

https://github.com/JetBrains/Exposed/wiki/DAO#many-to-many-reference

**Pull request closed without merging**